webapp/channels/src/components/threading/thread_viewer/thread_viewer.tsx

```diff
@@ -173,7 +174,21 @@ export default class ThreadViewer extends React.PureComponent<Props, State> {
     // scrolls to either bottom or new messages line
     private onInit = async (reconnected = false): Promise<void> => {
         this.setState({isLoading: !reconnected});
-        await this.props.actions.getPostThread(this.props.selected?.id || this.props.rootPostId, !reconnected);
+        const res = await this.props.actions.getPostThread(this.props.selected?.id || this.props.rootPostId, !reconnected, this.props.lastUpdateAt);
+
+        if (this.props.selected) {
+            const {order, posts} = res.data;
+            let highestUpdateAt = posts[order[0]].update_at;
+
+            // Check all posts to find the highest update_at
+            (Object.values(posts) as Post[]).forEach((post: Post) => {
```

Comment:
Github AI suggests we can use `for(const postId in posts)` ... Can we use that without object.values?

Reply:
That will give us the postIds, we want the posts. Are you suggesting to access the map again by the postId?

Reply:
Yes exactly

Reply:
Done